pandas/tests/io/test_parquet.py

```diff
@@ -0,0 +1,233 @@
+    for engine, have in [('pyarrow', _HAVE_PYARROW),
+                         ('fastparquet', _HAVE_FASTPARQUET)]))
+def engine(request):
+    return request.param
+
+
+@pytest.fixture
+def pa():
+    if not _HAVE_PYARROW:
+        pytest.skip("pyarrow is not installed")
+    return 'pyarrow'
+
+
+@pytest.fixture
+def fp():
+    if not _HAVE_FASTPARQUET:
+        pytest.skip("fastparquet is not installed")
+    return 'fastparquet'
+
+
+def test_invalid_engine():
+
+    with pytest.raises(ValueError):
+        tm.makeDataFrame().to_parquet('foo', 'bar')
+
+
+class Base(object):
+
+    def check_error_on_write(self, df, engine, exc):
+        # check that we are raising the exception
+        # on writing
+
+        with pytest.raises(exc):
+            with tm.ensure_clean() as path:
+                to_parquet(df, path, engine)
+
+    def check_round_trip(self, df, engine, expected=None, **kwargs):
+
+        with tm.ensure_clean() as path:
+            df.to_parquet(path, engine, **kwargs)
+            result = read_parquet(path, engine)
+
+            if expected is None:
+                expected = df
+            tm.assert_frame_equal(result, expected)
+
+            # repeat
+            to_parquet(df, path, engine, **kwargs)
+            result = pd.read_parquet(path, engine)
+
+            if expected is None:
+                expected = df
+            tm.assert_frame_equal(result, expected)
+
+
+class TestBasic(Base):
+
+    def test_error(self, engine):
+
+        for obj in [pd.Series([1, 2, 3]), 1, 'foo', pd.Timestamp('20130101'),
+                    np.array([1, 2, 3])]:
+            self.check_error_on_write(obj, engine, ValueError)
+
+    def test_stringify_columns(self, engine):
+
+        df = pd.DataFrame(np.arange(12).reshape(4, 3)).copy()
+        self.check_error_on_write(df, engine, ValueError)
+
+    def test_write_with_index(self, engine):
+
+        df = pd.DataFrame({'A': [1, 2, 3]})
+        self.check_round_trip(df, engine)
+
+        # non-default index
+        for index in [[2, 3, 4],
+                      pd.date_range('20130101', periods=3),
+                      list('abc'),
+                      [1, 3, 4],
+                      pd.MultiIndex.from_tuples([('a', 1), ('a', 2),
+                                                 ('b', 1)]),
+                      ]:
+
+            df.index = index
+            self.check_error_on_write(df, engine, ValueError)
+
+        # index with meta-data
+        df.index = [0, 1, 2]
+        df.index.name = 'foo'
+        self.check_error_on_write(df, engine, ValueError)
+
+        # column multi-index
+        df.index = [0, 1, 2]
+        df.columns = pd.MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1)]),
+        self.check_error_on_write(df, engine, ValueError)
+
+    @pytest.mark.parametrize('compression', [None, 'gzip', 'snappy', 'brotli'])
+    def test_compression(self, engine, compression):
+
+        if compression == 'snappy':
+            try:
```

Comment:
@jreback you can use an [importor skip](https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency) So I think

```python
pytest.importorskip("snappy")
``` 

instead of the try / except. I don't know if they can be used in fixture functions to handle the above `skipifs`.